src/components/trials/steps/FormStepDataSheet.tsx

```diff
@@ -263,7 +257,36 @@ function FormStepDataSheet<T>({
                     className={styles.datasheet}
                     data={processedGrid}
                     cellRenderer={CellRenderer}
-                    rowRenderer={RowRenderer}
+                    rowRenderer={React.memo(
+                        RowRenderer,
+                        // Block row re-renders if data in the row hasn't changed
+                        (oldProps: any, newProps: any) => {
+                            const notActive = !some(
+                                oldProps.children,
+                                ({ props: oldChildProps }, index) => {
+                                    const newChildProps =
+                                        newProps.children[index].props;
+                                    return (
+                                        oldChildProps.editing ||
+                                        oldChildProps.selected ||
+                                        newChildProps.editing ||
+                                        newChildProps.selected
+                                    );
+                                }
+                            );
+
+                            const noErrors = !some(
+                                oldProps.cells,
+                                ({ error: oldError }, index) => {
+                                    const newError =
+                                        newProps.cells[index].error;
+                                    return !!oldError || !!newError;
+                                }
+                            );
+
+                            return notActive && noErrors;
+                        }
+                    )}
```

Comment:
This is the key change in this PR yielding a performance boost. Block row re-renders except when a cell in that row has been selected / edited or has a validation error to display.